tools/terraform/fleetdm_client/fleetdm_client.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	url := teamPrefix + "/" + strconv.FormatInt(id, 10)
+	var teamUpdate TeamCreate
+	if name != nil {
+		teamUpdate.Name = *name
+	}
+	if description != nil {
+		teamUpdate.Description = *description
+	}
+	updateJson, err := json.Marshal(teamUpdate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update team body request: %w", err)
+	}
+	req, err := http.NewRequest(http.MethodPatch, url, bytes.NewReader(updateJson))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create PATCH request for %s body %s: %w",
+			url, updateJson, err)
+	}
+	resp, err := c.do(req, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to PATCH %s body %s: %w",
+			url, updateJson, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("failed PATCH %s body %s: %s %s",
+			url, updateJson, resp.Status, string(b))
+	}
+
+	var newTeam TeamGetResponse
+	err = json.NewDecoder(resp.Body).Decode(&newTeam)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &newTeam, nil
+}
+
+// UpdateAgentOptions pretends that the agent options are a string, when it's really actually json.
+// Strangely the body that comes back is a team, not just the agent options.
+func (c *FleetDMClient) UpdateAgentOptions(id int64, ao string) (*TeamGetResponse, error) {
+
+	// First verify it's actually json.
+	var aoJson interface{}
+	err := json.Unmarshal([]byte(ao), &aoJson)
+	if err != nil {
+		return nil, fmt.Errorf("agent_options might not be json: %s", err)
+	}
+
+	aoUrl := teamPrefix + "/" + strconv.FormatInt(id, 10) + "/" + "agent_options"
+	req, err := http.NewRequest(http.MethodPost, aoUrl, strings.NewReader(ao))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create agent_options POST request for %s id %d: %w",
+			teamPrefix, id, err)
+	}
+	resp, err := c.do(req, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to POST agent_options %s id %d: %w",
+			teamPrefix, id, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("failed to modify agent_options %d: %s %s", id, resp.Status, string(b))
+	}
+
+	var team TeamGetResponse
+	err = json.NewDecoder(resp.Body).Decode(&team)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode agent_options team response: %w", err)
+	}
+	return &team, nil
+}
+
+// DeleteTeam deletes the team with the provided ID.
+func (c *FleetDMClient) DeleteTeam(id int64) error {
+	url := teamPrefix + "/" + strconv.FormatInt(id, 10)
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create DELETE request for %s: %w", url, err)
+	}
+	resp, err := c.do(req, "")
+	if err != nil {
+		return fmt.Errorf("failed to DELETE %s: %w", url, err)
+	}
+
+	if resp.StatusCode >= 300 {
+		return fmt.Errorf("failed to delete team %d: %s", id, resp.Status)
+	}
+
+	defer resp.Body.Close()
```

Comment:
This should be moved after the `if err != nil` on line 265 above, so that it is closed even if `resp.StatusCode >= 300`.

Reply:
Oh, yep, good catch.